src/js/models/state.model.js

```diff
@@ -1,43 +1,79 @@
 export class StateModel {
-    state;
+    refreshState = {
+        restart: false,
+        reset: false,
+    };
+    numbersState = {
+        lastNumbers: [],
+        lastBonusNumber: [],
+        numberSet: [],
+        numberSetManuel: []
+    };
+    purchasedState = {
+        price: 0,
+        purchasedUnit: 0
+    };
     #observers = new Set();
 
-    constructor(state) {
-        this.state = state;
+    constructor() {
     }
 
     reset() {
-        this.state = {};
+        this.resetState();
         this.#observers = new Set();
     }
 
     resetState() {
-        this.state = {};
+        ['restart', 'reset'].forEach(prop => {
```

Comment:
(마이너) 실수할 여지를 줄여줄 수 있어요
```suggestion
        Object.keys(this.refreshState).forEach(prop => {
```